complete homework 1

Manisureal opened the pull request:
The following tasks are completed and included in this PR.

- Task 1.1: reverse input string
- Task 1.2: convert CSV and write into a txt file
- Task 1.3: rewrite the above using eS6


**How to run tasks locally?**
- Task1: npm run task1
- Task2: npm run task2
- Task3: 
  - To run task 1 in es6: npm run task1_es6
  - To run task 2 in es6: npm run task2_es6

manisureal-0 committed:
complete homework 1

Dmitry-White commented on src/task_1.1/task1.js:
```diff
@@ -0,0 +1,22 @@
+const process = require('process');
+const stdin = process.stdin;
```
Such variables could be considered redundant since they don't bring any additional meaning.
Moreover, it's best to avoid using reserved words as names for variables.

Dmitry-White commented on src/task_1.1/task1.js:
```diff
@@ -0,0 +1,22 @@
+  }
+});
+  
+stdin.on("end", () => {
```
Consider extracting the callback into a separate function.

Dmitry-White commented on src/task_1.2/task2.js:
```diff
@@ -0,0 +1,28 @@
+const csv = require('csvtojson');
+const { pipeline } = require('stream/promises');
+const path = require('path');
+const fs = require('fs');
```
It seems you're bringing the same package to the current file twise.
Consider importing everything you need from the package in one import statement.

Dmitry-White commented on src/task_1.2/task2.js:
```diff
@@ -0,0 +1,28 @@
+const fs = require('fs');
+const fsPromises = require('fs').promises;
+
+const readableStream = fs.createReadStream(path.join(__dirname, 'csv/nodejs-hw1-ex1.csv'));
```
Consider storing path to the files as constants at top of the file (after imports).

Dmitry-White commented on src/task_1.2/task2.js:
```diff
@@ -0,0 +1,28 @@
+const writableStream = fs.createWriteStream(path.join(__dirname, 'txt/nodejs-hw1-ex1.txt'));
+
+const writeConvertedCsvToTxtFile = async () =>  {
+  if (!fs.existsSync(path.join(__dirname, 'txt'))) {
```
Consider storing such expression results in a separate variable to improve readability.

Dmitry-White commented on src/task_1.2/task2.js:
```diff
@@ -0,0 +1,28 @@
+
+  await pipeline(
+    readableStream,
+    csv({
```
Consider creating a separate variable to store configs for any package.

Dmitry-White commented on src/task_1.3/task1_es6.js:
```diff
@@ -0,0 +1,25 @@
+  terminal: false
+});
+
+const reverseInput = () => {
```
Consider refactoring these functions to avoid recursion.
Additionally, consider always extracting callbacks into separate functions to improve readability.

Dmitry-White commented on src/task_1.3/task2_es6.js:
```diff
@@ -0,0 +1,30 @@
+import { join } from 'path';
+import { createReadStream, createWriteStream, existsSync } from 'fs';
+import { promises as fsPromises } from 'fs';
+
```
Same as for CommonJS version.

Dmitry-White reviewed (changes requested):
Looks good, left some comments.

manisureal-0 committed:
addressed homework_1 review comments

Manisureal commented:
> Looks good, left some comments.

@Dmitry-White Thank you, I have now addressed the review comments and are included with this [COMMIT](https://github.com/Manisureal/node-js-training/pull/1/commits/34a373c5744b1021d61f0d64676250e147369af2) - can you please re-review 👍

Dmitry-White reviewed (approved):
Good job!
